From: Miroslav Rezanina <mrezanin@redhat.com>
Date: Mon, 11 May 2009 08:35:23 -0400
Subject: [cpufreq] xen: powernow identifies wrong number of procs
Message-id: 217955372.137241242045323737.JavaMail.root@zmail06.collab.prod.int.phx2.redhat.com
O-Subject: Re: [PATCH RHEL 5.4] Fix BZ 456437 powernow identifies the wrong number of processors
Bugzilla: 456437
RH-Acked-by: Rik van Riel <riel@redhat.com>
RH-Acked-by: Brian Maly <bmaly@redhat.com>
RH-Acked-by: Chris Lalancette <clalance@redhat.com>

As is mentioned in comments for BZ 456437 (https://bugzilla.redhat.com/show_bug.cgi?id=456437), there is wrong info message during boot on machines with AMD processors. As Xen provides available cores into all doms (dom0 and domU) in form of single core virtual processors that has same model name as original processor. This can cause confusion as message
"Found X processors Model-name (Y cpu cores)"
looks like
"Found Y processors Model-name (Y cpu cores)"

To prevent this confusion, message format is changed into:
"Found Y virtualized processors"

--
Miroslav Rezanina
Software Engineer - Virtualization Team - XEN kernel

diff --git a/arch/i386/kernel/cpu/cpufreq/powernow-k8.c b/arch/i386/kernel/cpu/cpufreq/powernow-k8.c
index ce4f5e6..43b8b57 100644
--- a/arch/i386/kernel/cpu/cpufreq/powernow-k8.c
+++ b/arch/i386/kernel/cpu/cpufreq/powernow-k8.c
@@ -1469,6 +1469,7 @@ static int __cpuinit powernowk8_init(void)
 	}
 	if (powernow_k8_cpu_preinit_acpi())
 		printk(KERN_ERR PFX "Pre-initialization of ACPI failed\n");
+#ifndef CONFIG_XEN
 #ifdef CONFIG_SMP
 	printk(KERN_INFO PFX "Found %d %s "
 	       "processors (%d cpu cores) (" VERSION ")\n",
@@ -1479,6 +1480,10 @@ static int __cpuinit powernowk8_init(void)
 	       "processors (%d cpu cores) (" VERSION ")\n",
 	       boot_cpu_data.x86_model_id, supported_cpus);
 #endif
+#else
+	printk(KERN_INFO PFX "Found %d virtualized processors\n",
+		supported_cpus);
+#endif
 	return cpufreq_register_driver(&cpufreq_amd64_driver);
 }
 
